cmd/iam.go

```diff
@@ -991,13 +1022,29 @@ func (sys *IAMSys) IsAllowed(args iampolicy.Args) bool {
 		return sys.IsAllowedSTS(args)
 	}
 
-	sys.RLock()
-	defer sys.RUnlock()
+	if !args.IsOwner {
+		policies, err := sys.PolicyDBGet(args.AccountName, false)
+		if err != nil {
+			logger.LogIf(context.Background(), err)
+			return false
+		}
 
-	// If policy is available for given user, check the policy.
-	if mp, found := sys.iamUserPolicyMap[args.AccountName]; found {
-		p, ok := sys.iamPolicyDocsMap[mp.Policy]
-		return ok && p.IsAllowed(args)
+		// If policies were found, evaluate all of them.
+		if len(policies) > 0 {
```

Comment:
Perhaps return `if len(policies) == 0 { return false }` first?  to avoid indentation?